### PR TITLE
[PTQ] Update scales unification rules based on types

### PR DESCRIPTION
**Pull request by KodiaqQ**
### Changes

- Updated logic for scales unification based on the types

### Reason for changes

- Correctness of the unification functionality

### Related tickets

- 108258

### Tests

- Conformance run - 25

### CODE REVIEW

**Comment by kshpv** [comment]
Could you briefly explain why the previous approach was incorrect?

**Comment by KodiaqQ** [comment]
> Could you briefly explain why the previous approach was incorrect?

Unification should only take place for those quantizers that are `Concat` inputs but were generated by `Convolution` or `MatMul` layers with a per-tensor quantization scheme for activation.

File: nncf/common/quantization/quantizer_propagation/graph.py [hunk 1]
```diff
@@ -754,17 +754,40 @@ def get_paths_to_immediately_dominating_insertion_points(
...
+        self,
+        insertion_point_node_key: str,
+        unified_scale_op_metatypes: Set[Type[OperatorMetatype]],
+        unification_producing_metatypes: List[OperatorMetatype],
     ) -> Dict[Optional[int], List[PropagationPath]]:
         """Paths are lists of edges."""
         next_group_idx = 0
         paths = {}
 
+        def followed_by_weighted_types(curr_node_key) -> bool:
```
Comment (kshpv) [inline nncf/common/quantization/quantizer_propagation/graph.py:771]: The name confuses as there are no weighted types used.

**Review by kshpv — APPROVED** [review]

**Review by alexsu52 — CHANGES_REQUESTED** [review]
This PR required deep validation because FQs placment will be chaneged:
- e2e for all backends
- conformance test

**Comment by alexsu52** [comment]
> > This PR required deep validation because FQs placment will be chaneged:
> > 
> > * e2e for all backends
> > * conformance test
> 
> e2e pytorch job - No. 328 e2e tf job - No. 337 conformance job - No. 41

Could you provide a summary?

**Comment by KodiaqQ** [comment]
run pytorch pre-commit tests

**Review by alexsu52 — APPROVED** [review]
LGTM

**Pull request merged**